stdlib/2/os/__init__.pyi

```diff
@@ -144,66 +145,66 @@ def fsync(fd: int) -> None: ...
 def major(device: int) -> int: ...
 def minor(device: int) -> int: ...
 def makedev(major: int, minor: int) -> int: ...
-def mkdir(path: unicode, mode: int = ...) -> None: ...
-def makedirs(path: unicode, mode: int = ...) -> None: ...
-def pathconf(path: unicode, name: str) -> str: ...
+def mkdir(path: _PathType, mode: int = ...) -> None: ...
+def makedirs(path: _PathType, mode: int = ...) -> None: ...
+def pathconf(path: _PathType, name: str) -> str: ...
 def readlink(path: AnyStr) -> AnyStr: ...
-def remove(path: unicode) -> None: ...
-def removedirs(path: unicode) -> None: ...
-def rename(src: unicode, dst: unicode) -> None: ...
-def renames(old: unicode, new: unicode) -> None: ...
-def rmdir(path: unicode) -> None: ...
-def stat(path: unicode) -> Any: ...
+def remove(path: _PathType) -> None: ...
+def removedirs(path: _PathType) -> None: ...
+def rename(src: _PathType, dst: _PathType) -> None: ...
+def renames(old: _PathType, new: _PathType) -> None: ...
+def rmdir(path: _PathType) -> None: ...
+def stat(path: _PathType) -> Any: ...
 @overload
 def stat_float_times(newvalue: bool = ...) -> None: ...
 @overload
 def stat_float_times() -> bool: ...
-def statvfs(path: unicode) -> _StatVFS: ...
-def symlink(source: unicode, link_name: unicode) -> None: ...
-def unlink(path: unicode) -> None: ...
-def utime(path: unicode, times: Optional[Tuple[int, int]]) -> None: ...
-def walk(top: AnyStr, topdown: bool = ..., onerror: Any = ...,
-         followlinks: bool = ...) -> Iterator[Tuple[AnyStr, List[AnyStr],
-                                                    List[AnyStr]]]: ...
+def statvfs(path: _PathType) -> _StatVFS: ...
+def symlink(source: _PathType, link_name: _PathType) -> None: ...
+def unlink(path: _PathType) -> None: ...
+def utime(path: _PathType, times: Optional[Tuple[int, int]]) -> None: ...
+def walk(top: _PathType, topdown: bool = ..., onerror: Any = ...,
+         followlinks: bool = ...) -> Iterator[Tuple[_PathType, List[_PathType],
+                                                    List[_PathType]]]: ...
 
 def abort() -> None: ...
-def execl(file: AnyStr, *args) -> None: ...
-def execle(file: AnyStr, *args) -> None: ...
-def execlp(file: AnyStr, *args) -> None: ...
-def execlpe(file: AnyStr, *args) -> None: ...
-def execv(path: AnyStr, args: Union[Tuple[AnyStr], List[AnyStr]]) -> None: ...
-def execve(path: AnyStr, args: Union[Tuple[AnyStr], List[AnyStr]], env: Mapping[AnyStr, AnyStr]) -> None: ...
-def execvp(file: AnyStr, args: Union[Tuple[AnyStr], List[AnyStr]]) -> None: ...
-def execvpe(file: AnyStr, args: Union[Tuple[AnyStr], List[AnyStr]], env: Mapping[AnyStr, AnyStr]) -> None: ...
+def execl(file: _PathType, *args) -> None: ...
+def execle(file: _PathType, *args) -> None: ...
+def execlp(file: _PathType, *args) -> None: ...
+def execlpe(file: _PathType, *args) -> None: ...
+def execv(path: _PathType, args: Union[Tuple[_PathType], List[_PathType]]) -> None: ...
```

Comment:
PathType is an unfortunate name for the args, since they don't have to be Paths.

I think the best solution might be to add a second type alias (or just use Union directly here), so we can easily change the _PathType alias later to include os.PathLike in Python 3.6+.

This also applies to the other exec* function and to the spawn* functions below.